From e458f32827c26df5767da9a61c42d8465c68062d Mon Sep 17 00:00:00 2001
From: Zhe Wang <zhe.wang@amlogic.com>
Date: Thu, 27 Aug 2015 15:14:37 +0800
Subject: [PATCH 5793/5965] PD#111655: audio: reinit audio codec in resume

instaboot on TV platform, pmu4 and tas5707 must be reinited in resume.

Change-Id: I2f0738add624174544ac6ebe488e47d4d15e8af7
---
 sound/soc/codecs/aml_pmu4_codec.c | 6 ++++++
 sound/soc/codecs/tas5707.c        | 2 +-
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/sound/soc/codecs/aml_pmu4_codec.c b/sound/soc/codecs/aml_pmu4_codec.c
index c749ff5e1e6f..3c58de2d40da 100644
--- a/sound/soc/codecs/aml_pmu4_codec.c
+++ b/sound/soc/codecs/aml_pmu4_codec.c
@@ -478,6 +478,12 @@ static int aml_pmu4_audio_suspend(struct snd_soc_codec *codec)
 
 static int aml_pmu4_audio_resume(struct snd_soc_codec *codec)
 {
+    printk("enter %s\n",__func__);
+    aml_pmu4_audio_power_init();
+    aml_pmu4_audio_reset(codec);
+    aml_pmu4_audio_start_up(codec);
+    aml_pmu4_audio_reg_init(codec);
+    codec->dapm.bias_level = SND_SOC_BIAS_STANDBY;
     aml_pmu4_audio_set_bias_level(codec, SND_SOC_BIAS_STANDBY);
     return 0;
 }
diff --git a/sound/soc/codecs/tas5707.c b/sound/soc/codecs/tas5707.c
index 6e9ebddaaa91..b7d584df0f77 100644
--- a/sound/soc/codecs/tas5707.c
+++ b/sound/soc/codecs/tas5707.c
@@ -357,7 +357,7 @@ static int tas5707_set_eq(struct snd_soc_codec *codec)
 	if(!pdata)
 		return -ENOENT;
 
-	if(pdata->num_eq_cfgs){
+	if (pdata->num_eq_cfgs && (tas5707->eq_conf_texts == NULL)) {
 		struct snd_kcontrol_new control =
 			SOC_ENUM_EXT("EQ Mode", tas5707->eq_conf_enum, tas5707_get_eq_enum, tas5707_put_eq_enum);
 
-- 
2.19.0

